cypress/e2e/facility_spec/FacilityHomepage.cy.ts

```diff
@@ -181,6 +221,99 @@ describe("Facility Homepage Function", () => {
+    });
+
+    // create a new patient in the facility
+    cy.visit("/patients");
+    patientPage.createPatientWithData(newPatientData);
+    // navigate to facility page and verify the occupancy badge
+    cy.visit("/facility");
+    manageUserPage.typeFacilitySearch(facilityWithNoAvailableBeds);
+    facilityPage.verifyFacilityBadgeContent(facilityWithNoAvailableBeds);
+    facilityHome.verifyOccupancyBadgeVisibility();
+    manageUserPage.assertFacilityBadgeContent("1", "0");
+    manageUserPage.assertFacilityBadgeBackgroundColor("rgb(239, 68, 68)");
+    // create a new location and add a bed to the facility
+    facilityPage.visitAlreadyCreatedFacility();
+    cy.get("[id='manage-facility-dropdown']").scrollIntoView().click();
+    cy.get("[id=location-management]").click();
+    // create new location and add a bed to the facility
+    cy.get("body").then(($body) => {
+      if ($body.find("#manage-bed-button").length > 0) {
+        facilityLocation.clickManageBedButton();
+      } else {
+        facilityLocation.clickAddNewLocationButton();
+        facilityPage.fillFacilityName(locationName);
+        facilityLocation.selectLocationType(locationType);
+        assetPage.clickassetupdatebutton();
+        facilityLocation.clickNotification();
+        facilityLocation.clickManageBedButton();
+      }
+    });
+    facilityLocation.clickAddBedButton();
+    facilityLocation.addBed("Bed 1", "Test Description", "Regular", 2);
+    // navigate to patient page, and click create consultation
+    cy.visit("/facility");
+    manageUserPage.typeFacilitySearch(facilityWithNoAvailableBeds);
+    facilityPage.verifyFacilityBadgeContent(facilityWithNoAvailableBeds);
+    // visit facility patients page
+    manageUserPage.clickFacilityPatients();
+    facilityHome.verifyPatientListVisibility();
+    facilityHome.verifyPatientListUrl();
+    // type patient name and click create consultation
+    patientPage.visitPatientWithNoConsultation(newPatientData.name);
+    // create patient consultation and add bed to the consultation
+    patientConsultationPage.selectConsultationStatus(
+      "Outpatient/Emergency Room",
+    );
+    cy.get("#is_asymptomatic").click();
+    patientConsultationPage.selectPatientCategory("Mild");
+    patientConsultationPage.typePatientNumber(patientIpNumber);
+    patientConsultationPage.selectPatientDiagnosis(
+      "1A00",
+      "add-icd11-diagnosis-as-unconfirmed",
+    );
+    patientTreatmentPlan.fillTreatingPhysician(doctorName);
+    patientConsultationPage.selectBed("Bed 1");
+    cy.clickSubmitButton("Create Consultation");
+    cy.verifyNotification("Consultation created successfully");
+    // verify the occupancy badge reflection
+    cy.visit("/facility");
+    manageUserPage.typeFacilitySearch(facilityWithNoAvailableBeds);
+    facilityPage.verifyFacilityBadgeContent(facilityWithNoAvailableBeds);
+    facilityHome.verifyOccupancyBadgeVisibility();
+    manageUserPage.assertFacilityBadgeContent("1", "2");
```

Comment:
- this badge is part of facility card so keep the new function in facilityHome.ts